crtclient/crtclient.go

```diff
@@ -15,29 +15,26 @@ limitations under the License.
 
 package crtclient
 
-// container runtime client interface implementations
-
-type Config struct {
-	// Config object parsed from a json styled config
-	V interface{}
-}
-
-type ContainerEpContext struct {
+// ContainerEPContext is used to manage the parameters for all
+// container+endpoint management operations.
+type ContainerEPContext struct {
 	NewContName    string
 	NewAttachUUID  string
 	CurrContName   string
 	CurrAttachUUID string
-	InterfaceId    string
-	IpAddress      string
+	InterfaceID    string
+	IPAddress      string
 	SubnetLen      uint
 	DefaultGw      string
 }
 
+// ContainerIf implementations are used to configure and manage container
+// interfaces.
 type ContainerIf interface {
-	Init(config *Config) error
+	Init(config interface{}) error
```

Comment:
This was not, I should have made note of it or just flat-out not included it in this patch, sorry. I will probably remove it, but let’s sort this out.

I presume you want to ensure it’s a JSON object? Is there some reason you cannot just `type Config interface{}`?

> On May 26, 2015, at 10:26 AM, Madhav Puri <notifications@github.com> wrote:
> 
> In crtclient/crtclient.go <https://github.com/contiv/netplugin/pull/82#discussion_r31057885>:
> 
> >  type ContainerIf interface {
> > -	Init(config *Config) error
> > +	Init(config interface{}) error
> Just curious, was this reported by go-lint? Config was defined as a struct just so as to make it more explicit that it contains an interface consumable by json marshal/unmarshal routines. Albeit we could have used core.Config here itself than defining a new struct.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/contiv/netplugin/pull/82/files#r31057885>.
>